torch/cuda/amp/autocast_mode.py

```diff
@@ -113,11 +113,8 @@ def forward(self, input):
         enabled(bool, optional, default=True):  Whether autocasting should be enabled in the region.
     """
     def __init__(self, enabled=True):
-        if enabled and not torch.cuda.is_available():
-            warnings.warn("torch.cuda.amp.autocast only affects CUDA ops, but CUDA is not available.  Disabling.")
-            self._enabled = False
-        else:
-            self._enabled = enabled
+        # Disable the cuda device checking to reuse the code in torch/xla
+        self._enabled = enabled
```

Comment:
I'm pretty opposed to removing the logic on the left.  The logic allows people to take a script written for amp on the GPU and run it on the CPU without having to rewrite their script at all (GradScaler disables itself automatically with a helpful message).

Why do you need to remove the logic on the left?  If you're running Amp, presumably you're targeting GPUs (TPUs don't support fp16) so there must be GPUs on the system.

Reply:
Because it is possible that we could use cpu-version pytorch and train on GPUs with torch/xla.

Reply:
I added a `skip_device_check` argument here. Then the code in torch/xla could skip cuda check. Do you think it is acceptable?

Reply:
I'm not a fan of adding to the user-facing API unless we really have to.

Are you saying the case we need to enable is:
- Pytorch was built for CPU only
- the training script runs with an xla tensors, and under the hood, XLA targets GPUs?

Is there a way to tell if xla is targeting GPUs?  If so we could replace
```
if not skip_device_check and enabled and not torch.cuda.is_available():
```
with
```
if enabled and not torch.cuda.is_available() and not xla_is_targeting_gpus():
```
and remove the need for a public argument.

Reply:
- Pytorch was built for CPU only
- the training script runs with an xla tensors, and under the hood, XLA targets GPUs?
-----------------
Yeah, this is exactly what I mean.


And as torch/xla identified the devices by the environment variable: `XRT_DEVICE_MAP`, an example of `XRT_DEVICE_MAP` is `GPU:0;/job:localservice/replica:0/task:0/device:XLA_GPU:0`. Could I define `xla_is_targeting_gpus" as below:

```
def xla_is_targeting_gpus():
    if os.getenv("XRT_DEVICE_MAP"):
        return "GPU" in os.getenv("XRT_DEVICE_MAP")
    return False
```

Reply:
Anything that provides the `xla_is_targeting_gpus()` interface to enable
```python
if enabled and not torch.cuda.is_available() and not xla_is_targeting_gpus():
```
is fine for me.  I know almost nothing about XLA so I'm not sure the XRT_DEVICE_MAP check is the right approach, but it's an implementation detail you could change later.